docs/requirements/path-unwinding-forwarding -requirements.md

```diff
@@ -0,0 +1,98 @@
+
+### 2. Transferring liquid staking derivatives
+
+Liquid staking derivatives are minted on liquid staking zones and represent a staked asset. There is a common misconception from users that these derivatives originate on the chain of the original staked token. This results in users sending derivatives back to the chain of the natively staked token and then onto the next destination. 
+
+For examples, a user has stATOM on Osmosis, they want to move the stATOM to Evmos, instead of going from Osmosis --> Stride --> Evmos, a user tries to unwind themself and routes the tokens Osmosis --> Cosmos Hub --> Evmos. 
+
+### 3. Moving a token that originated from an interoperability zone or chain, or asset issuer
+
+Tokens that originate from other blockchain ecosystems that don't yet support IBC, flow into the Cosmos ecosystem through interoperability zones. These tokens are then sent onto other chains with a specific use case for these tokens and a user could want to move this token from one chain to another. 
+
+For example, ETH from Ethereum flows into Osmosis via Axelar, where the final step moving ETH from Ethereum to Osmosis uses an ICS-20 transfer from Axelar to Osmosis. A user may then want to move ETH from Osmosis onto another chain, for example Injective. However, the path with 1 hop would be a transfer from Axelar to Injective.
+
+### 4. Moving a token from one chain to another, swapping the token and transferring it onwards to a new destination
+
+A user on one chain, for example the Cosmos Hub holds an asset, e.g. ATOM and wants to instead have AKT on Akash. The user must transfer to ATOM to a DEX chain, swap the ATOM for AKT and then send the AKT onwards to Akash. 
+
+# Functional requirements
+
+## Assumptions and dependencies
+
+1. A functional relayer implementation is required for this feature.
+2. Routing information for the final hop for unwinding or for forwarding is configured for the user through a front end client, or configured by another means using off-chain data. 
+3. The feature will have no impact on the existing function of a typical ICS-20 transfer where a native token is sent to another chain.
+4. Fees will be treated the same as with other IBC applications.
+5. The functionality to enable a specific action before forwarding is not in scope of these requirements.
+6. If a transfer contains multiple unique tokens, the unwinding functionality only needs to support a single denom, i.e. if there is a transfer containing a native token and 1 hop denom being sent from source to destination, unwinding should support unwinding the 1 hop denom and sending the native token directly to the destination. Support for transfer and unwind for native token and 2+ 1 hop denoms is not required.
```

Comment:
Checking if I understood this correctly:
- If the transfer contains more that one denom (native and/or with >1 hops)  and the user wants to do unwinding + forwarding, should we return an error and let the user know that the transfer should include only one denom?
- If the transfer contains more that one denom (native and/or with >1 hops) and the user wants to do forwarding, should we just forward all denoms through the same forwarding path?

Reply:
apologies for the delay, completely missed the notification

My thinking is that we want to be able to support the use case of transfer + supply liquidity and typically pools are 2 sided, maybe we could stretch to 3 asset pools but I think 2 asset pools should be sufficient to support initially. Therefore what I was hoping is that if you are on chain A, with token A and token C going to chain B, you would be able to send A and C to B, but C will route back from A -> C -> B so when it arrives it is still a 1 hop denom. I was hoping that you would be able to specify the forwarding path for up to 2 denoms but more its likely getting quite chaotic and messy I would assume and you only really need 2 typically for transfer + supply liquidity.

so in case 1, it should be ok with up to 2 different denoms. 
case 2, the forwarding path should be specified per denom

happy to make some changes to clarify this

Reply:
Yeah, I think we need some more discussion around this, because the spec currently supports only forwarding on a single path (i.e. if the transfer contains multiple coins, all of the coins will be forwarded through the same path; there is no possibility at the moment to specify a different forwarding path per coin).

Just superficially thinking about it: I guess if we want to support that we would need to send multiple packets (one for each coin that takes a different route), but then it could get complicated to acknowledge the original packet, since we would have to aggregate the acknowledgments/timeouts for each of those packets. And wouldn't things get also hairy if some packets succeed and some don't?